Tiltfile

```diff
@@ -1,31 +1,115 @@
+def printup(*args):
+  if config.tilt_subcommand == "up":
+    print(*args)
+
+config.define_bool('local-kibana')
+
+parsed_config = config.parse()
+
+script_dir = os.path.join(config.main_dir, 'script')
+
+default_kibana_port = 5601
+kibana_host = "localhost"
+kibana_base_path = ""
+
+# checks whether (dev mode) kibana is already running at given port,
+# and returns the URL it is redirected to
+def discover_kibana():
+
+    default_kibana_host = "localhost"
+
+    count = int(
+      str(
+        local("lsof -n -i :%s | grep LISTEN | wc -l | tr -d ' '" % default_kibana_port, quiet=True, echo_off=True)
+      ).strip()
+    )
+
+    if count <= 0:
+      fail("No service found on port " + str(default_kibana_port))
+      return None
```

Comment:
Is this discovery logic still needed, now that we're explicitly saying to use a local kibana? I'd rather like to drop it if it's not necessary.

Reply:
This is needed, still, to get the redirect URL (which includes the kibana base path).

Reply:
The curl is needed, but the lsof bit above? It's second guessing the flag. Could we just have the curl request fail if kibana isn't running?

Reply:
IMHO I think it's a good thing, the flag doesn't control the service, an explicit failure is still helpful here, especially because curl doesn't really give user-friendly error messages. If you feel strongly about it I can remove it.

Reply:
OK, let's leave it for now. Just trying to keep this thing minimal and maintainable :)
Should be fine.